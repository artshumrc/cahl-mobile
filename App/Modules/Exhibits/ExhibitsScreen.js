import React from 'react';
import PropTypes from 'prop-types';
import { ScrollView, View, Text } from 'react-native';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import I18n from 'react-native-i18n';

// components
import Exhibit from './Exhibit.js';

// styles
import styles from './ExhibitsScreenStyles';

// translations
import en from '../../i18n/languages/english.json';
import fr from '../../i18n/languages/fr.json';

I18n.translations = {
  en,
  fr,
};

class ExhibitsScreen extends React.Component {
  constructor(props) {
    super(props);

    this.navigateToComments = this.navigateToComments.bind(this);
  };

  static propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func,
    }).isRequired,
    data: PropTypes.shape({
      loading: PropTypes.bool.isRequired,
      exhibits: PropTypes.object,
    }).isRequired,
  };

  navigateToComments() {
    const { navigate } = this.props.navigation;
    navigate('CommentsScreen');
  }

  render() {
    const { navigation, data: { loading, error, HULItems } } = this.props;

    if (loading) {
      return (
        <Text style={styles.title}>{I18n.t('loading')}</Text>
      )
    } else if (error) {
      console.log(error);
      return (
        <Text style={styles.title}>{I18n.t('error')}</Text>
      )
    } else {
      const totalNumberOfItems = HULItems.pagination.numFound;
      const exhibits = HULItems.items.mods;

      return (
        <ScrollView style={styles.container}>
          <View style={styles.textBox}>
            <Text style={styles.title}>{I18n.t('charlieArchive')}</Text>
            <Text style={styles.subtitle}>{I18n.t('atHUL')}</Text>
          </View>
          <View>
            {
              exhibits.map(exhibit =>
                <Exhibit
                  exhibitNumber={exhibits.indexOf(exhibit) + 1}
                  navigation={navigation}
                  key={exhibit.recordInfo.recordIdentifier['#text']}
                  totalNumberOfItems={totalNumberOfItems}
                  imageSource={exhibit.relatedItem.location[0].url[0]['#text']}
                  description={exhibit.subject.topic}
                  recordId={exhibit.recordInfo.recordIdentifier['#text']}
                />
              )
            }
          </View>
        </ScrollView>
      );
    }
  }
}

const getExhibits = gql`
query {
  HULItems {
    pagination,
    items
  }
}
`;

export default graphql(getExhibits)(ExhibitsScreen);
