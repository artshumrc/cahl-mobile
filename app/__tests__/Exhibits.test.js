import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ExhibitsScreen from '../modules/Exhibits';

it('renders correctly', () => {
  const tree = renderer.create(
    <ExhibitsScreen />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
