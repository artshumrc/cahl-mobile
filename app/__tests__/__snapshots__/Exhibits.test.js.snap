// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "justifyContent": "space-around",
          "marginBottom": 100,
          "marginLeft": 15,
          "marginTop": 100,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": "AvenirNext-heavy",
            "fontSize": 34,
          }
        }
      >
        Charlie Archive
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#999999",
            "fontFamily": undefined,
            "fontSize": 19,
          }
        }
      >
        at the Harvard Library
      </Text>
    </View>
    <View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "AvenirNext-ultralight",
              "fontSize": 16,
              "marginLeft": 15,
            }
          }
        >
          1/265
        </Text>
        <Image
          source={
            Object {
              "uri": "/Users/tyler/Archimedes/cahl-mobile/App/images/#JeSuisCharlie.png",
            }
          }
          style={
            Object {
              "height": 500,
              "width": 750,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "AvenirNext-regular",
                "fontSize": 14,
                "marginLeft": 15,
              }
            }
          >
            A brief description might accompany the piece to describe it's meaning. If it is longer than three lines, the description would fold fold fold fold fold fold fold fold fold fold fold fold
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "color": "#999999",
                "fontFamily": "AvenirNext-regular",
                "fontSize": 14,
              }
            }
          >
              More
          </Text>
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": 15,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999999",
                "fontFamily": "AvenirNext-regular",
                "fontSize": 14,
              }
            }
          >
            View 2 comments
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
